File: lec01/optional/optional.py
# run with: `python3 optional.py`

def create(b):
    if b:
        return "Hello there"
    else:
        return None

def main():
    # Method 1
    create_true = create(True)
    if create_true:
        print("create(True) returned " + create_true)

    # Method 2
    create_false = create(False)
    print(
        "create(False) returned "
        + create_false if create_false else "<empty>"
    )

if __name__ == "__main__":
    main()

File: lec01/optional/optional.rs
// build with: rustc optional.rs

fn create(b: bool) -> Option<&'static str> {
    if b {
        Some("Hello There")
    } else {
        None
    }
}

fn main() {
    // method 1
    let create_true = create(true);
    if create_true.is_some() {
        println!("create(true) returned {}", create_true.unwrap());
    }

    // method 2
    println!(
        "create(false) returned {}",
        create(false).unwrap_or("<empty>")
    );








    // ...
    // ...
    // ...














    // method 3
    let create_true = create(true);
    match create_true {
        Some(string) => println!("create(true) returned {string}"),
        None         => println!("create(false) returned <empty>"),
    }
}
